common/persistence/cassandra/queue_v2_store.go

```diff
@@ -181,21 +196,75 @@ func (q *queueV2Store) ReadMessages(
 	}, nil
 }
 
-func (q *queueV2Store) getMinMessageID(request *persistence.InternalReadMessagesRequest) (int, error) {
-	// TODO: start from the ack level of the queue partition instead of the first message ID when there is no token.
-	if len(request.NextPageToken) == 0 {
-		return persistence.FirstQueueMessageID, nil
+func (q *queueV2Store) CreateQueue(
+	ctx context.Context,
+	request *persistence.InternalCreateQueueRequest,
+) (*persistence.InternalCreateQueueResponse, error) {
+	queueType := request.QueueType
+	queueName := request.QueueName
+	queue := persistencespb.Queue{
+		Partitions: map[int32]*persistencespb.QueuePartition{
+			0: {
+				MinMessageId: persistence.FirstQueueMessageID,
+			},
+		},
+	}
+	bytes, _ := queue.Marshal()
+	applied, err := q.session.Query(
+		TemplateCreateQueueQuery,
+		queueType,
+		queueName,
+		bytes,
+		enums.ENCODING_TYPE_PROTO3.String(),
+		0,
+	).WithContext(ctx).MapScanCAS(make(map[string]interface{}))
+	if err != nil {
+		return nil, gocql.ConvertError("QueueV2CreateQueue", err)
+	}
+
+	if !applied {
+		return nil, fmt.Errorf(
+			"%w: queue type %v and name %v",
+			persistence.ErrQueueAlreadyExists,
+			queueType,
+			queueName,
+		)
+	}
+	return &persistence.InternalCreateQueueResponse{}, nil
+}
+
+func (q *queueV2Store) getMinMessageID(
+	nextPageToken []byte,
+	queue *persistencespb.Queue,
+) (int, error) {
+	if len(nextPageToken) == 0 {
+		// Currently, we only have one partition for each queue. However, that might change in the future. If a queue is
+		// created with more than 1 partition by a server on a future release, and then that server is downgraded, we
+		// will need to gracefully handle this case. One way is to return the smallest possible message ID and log a
+		// warning. Since this function is just used as a shortcut to skip over tombstones, all the logic will still be
+		// correct. It just won't be as efficient. Then, the user can upgrade their server to a version which does have
+		// the ability to handle multiple partitions, and the warning will go away.
+		if len(queue.Partitions) != 1 {
```

Comment:
I think we should just return error here. The read request might be for a different partition and we will always return messages from partition 0 which is very confusing.